### PR TITLE
Version 1.6.2 Beta

### LINKED ISSUE #139: Live profile mouse interaction feature
From https://github.com/razman786/tigger_py5/issues/156#issuecomment-815916852

To have a profile that changes while you hover, clicking then freezes a profile.

### PR DESCRIPTION
**Pull request by razman786**
Draft PR for version 1.6.2:

Closes #139, #140 and #141

- Version bump in preparation for release
- Adds selected profile feature (ALT+LeftButton)
- WCS projection is now optimised for multiple images
- WCS projection provides 2 modes, switchable via the new WCS Projection menu
- Adds FITS header and WCS projection viewer
- Coordinate measurements have been fixed
- Fixes plot layout handling
- Fixes dockable widget sizing and placement
- Fixes historgram mouse wheel movement
- Fixes issue #139 - Live and selected profiles
- Fixes issue #140 - WCS projection
- Fixes issue #141 - flake8 errors
- Various other minor refactorings and adjustments

N.B. There are a number of changes in this version.

### CODE REVIEW

**Comment by razman786** [comment]
Some screenshots:

FITS Header viewer
1_Header_viewer

WCS menu
2_menu

Projection - Mean reference coordinate
3_proj_mean_ref_coord

Projection - Image reference coordinate
4_proj_image_ref_coord

Ruler measurements
5_ruler_measure_1
6_ruler_measure_2
7_ruler_measure_3

Profile selection
8_profile_selected
9_profile_selected_2

**Comment by razman786** [comment]
Unfortunately, my workflow software setup had a bug, so some commits were messed up. Cleaned with a cardinal sin.

**Comment by bennahugo** [comment]
ok to test

**Comment by bennahugo** [comment]
Thanks @razman786 this looks great. I will do some detailed testing hopefully tomorrow some time!

**Comment by razman786** [comment]
> Should this be formally reviewed - I see it is marked draft. I've done a wee bit of local testing last week and it looks well behaved. Haven't tested it with multiple mosaic pointings as of yet. Let me know when I should start that

I left it as a draft for a few reasons: 

1. The WCS projection of multiple images needs checking (and measurements), along with the use of the new 'mean centre reference coordinate' (personally I like it, but that's just me). N.B. It may flag an error with some combos of images, but this seems to be an issue raised from `reproject` - yet to be solved in those corner cases (I suspect some wrapping somewhere, or something I have overlooked, or it is simply a limitation).
2. The workflow of selecting profiles (with ALT+LeftButton) needs comment. Does it meet the issues spec correctly for selecting a single spectra? Ultimately, this becomes the base for tracking multiple spectra, so I want to know if this is OK before going further.
3. I have seen a small bug that seems to present itself every now and then with multiple images loaded and taking measurements. Seems like (although I haven't had a moment to get to the bottom of it) the extents used in the main plotting, cut off a corner of the image projection in-line with the grid circles being drawn (i.e. when an image is in the bottom left quadrant), so some measurements are incorrect in this corner case, notability so.

Basically, I need comment on these 3 items to get it ready for review. Any help is much appreciated! \o/

**Comment by bennahugo** [comment]
Looks good as discussed -- the biggest issue we need another button for alt click -- does not work out of the box with KDE, as it is the combo to move windows.... Perhaps we should just make this click when the fixed profile is active?

**Comment by bennahugo** [comment]
@o-smirnov @IanHeywood ok I'm reasonably happy with my addition to the static "freezed" selected profile mechanism now. Please review. 
We now have something like this (toggled by F4):
image
Features
- As many individual  selected profiles as you want
- Saving / restoring to and from disk through a "TigProf" file
- Overlays (to your choice of colour) such as
image

(profile names can be set arbitrarily)

Currently I only overlay static profiles (very similar to DS9 in other words with its copy and paste functionality). I was thinking of adding dynamically linked profiles where if you overlay profile a from profile b and you update profile a the overlay is also updated. However, I think as a first stab at this a static overlay for comparison is probably already good enough?

The only other feature that Oleg requested is H and V lines through the image if you have a selected profile on X or Y.
